cns/service/main.go

```diff
@@ -642,7 +642,13 @@ func main() {
 			return nil
 		}),
 	}
-	go healthserver.Start(z, cnsconfig.MetricsBindAddress, &healthz.Handler{}, readyChecker)
+
+	healthzHandler, err := healthserver.NewHealthzHandlerWithChecks(cnsconfig)
```

Comment:
If I'm reading the code correctly, this code is not behind any flags and will fail if there's no available kubeconfig/can't create an apiserver client. CNS does not always run in k8s nodes and this would be a breaking change for non-k8s scenarios. Please put this behind some flag.

Reply:
it is only going to try and read from kubeconfig when the `ChannelMode` is `CRD`. Is that ever set for non k8s scenarios?

Reply:
Usually set to `direct` for when DNC is in the same vnet as CNS and can ping it directly. This constructor (`NewHealthzHandlerWithChecks`) errors if it can't build a controller-runtime client though, no?

Reply:
yes, you're right my bad. forgot I put the client initialization outside the CRD block. updated in this commit https://github.com/Azure/azure-container-networking/pull/3269/commits/f4c50b565c27e19cde378a47e1df113bfd2332c8 ... I still feel like using ChannelMode CRD is safe here since it's already being used to fall into this section (https://github.com/Azure/azure-container-networking/blob/281770bf9714eba4e6b1fe413fe0fb253226d450/cns/service/main.go#L819) which calls `InitializeCRDState` which unconditionally tries to load a kubeconfig the same way we're doing here.